server/src/proxyManagement.mjs

```diff
@@ -21,97 +21,114 @@
+        // add ws connection to map
+        wsMap.set(returnWs, ws);
+        res(ws);
       });
 
       ws.on('close', function close() {
         console.log('disconnected');
-      });
-      
-      ws.on('message', function message(data) {
-        const buf = Buffer.from(data, 'utf8');
-        //send response to callback
-        callback(buf.toString(), returnWS)
+        // remove closed connection from map
+        wsMap.delete(returnWs);
       });
 
       ws.on('error', function message(err) {
-        rej(err)
+        rej(err);
       });
-    })
+    });
   } catch (err) {
-    return err
+    return err;
   }
 }
 
-/* Consume response from server. In case of event, send the event to caller directly.
-In case of response for requested method, store it in responseObject array
-*/
-function actOnResponseObject(data, returnWS) {
-  const response = JSON.parse(data)
-  if(response.id === undefined) {
-    returnWS.send(data)
-  }
-  responseObject[response.id] = data
-}
-
-function sendRequest(command) {
-  if(ws) {
-    ws.send(command)
-    console.log("Request sent to proxy server: ", command);
-  } else {
-    console.log("WS Client not initialized. Unable to send request to proxy server: ", command);
-  }
-}
-
-//Poll for proxy response. Fetch response using requestId. If timedout, terminate the connection
-function getResponseMessageFromProxy(id) {
-  let timeout = 2000
-  let counter = 0
-  let interval = 100
-  return new Promise((resolve, reject) => {
-    var timer = setInterval(function() {
-      if(counter >= timeout) {
-        console.log("response not received for given id: " + id)
-        reject(false)
+function getResponseMessageFromProxy(returnWs) {
+  let timeout = 10000;
+  let counter = 0;
+  let interval = 100;
+  return new Promise((res, rej) => {
+    var timer = setInterval(function () {
+      if (counter >= timeout) {
+        console.log('response not received for given returnWs');
+        rej(false);
       }
-      if(responseObject[id]) {
-        counter = timeout + interval
-        //clear interval if response received for given id.
-        clearInterval(timer)
-        resolve(responseObject[id])
+
+      const returnMsg = wsMsgMap.get(returnWs);
+      if (returnMsg) {
+        counter = timeout + interval;
+        //clear interval if response received for given returnWs.
+        clearInterval(timer);
+        res(returnMsg);
       }
-      counter = counter + interval
+      counter = counter + interval;
     }, interval);
-  })
-  
+  });
 }
 
 function buildWSUrl() {
-  let proxyUrl = process.env.proxyServerIP
-  if( ! proxyUrl ) {
-    throw Error('ERROR: Proxy Url not found in env')
-  } else if ( ! proxyUrl.includes(":") ) {
-    proxyUrl = proxyUrl + ":" + 9998
+  let proxyUrl = process.env.proxyServerIP;
+  if (!proxyUrl) {
+    throw Error('ERROR: Proxy Url not found in env');
+  } else if (!proxyUrl.includes(':')) {
+    proxyUrl = proxyUrl + ':' + 9998;
```

Comment:
I know this wasn't a part of the original ticket but let's just throw a log here saying we're using the default 9998 port

Reply:
Add additional console log